library/alloc/src/raw_vec.rs

```diff
@@ -498,7 +498,7 @@ impl<A: Allocator> RawVecInner<A> {
     }
```

Comment:
Mark `fn non_null` `const`, just replace `.into()` with `.as_non_null_ptr()` ([source](https://github.com/rust-lang/rust/blob/e6eb45143cc4dd236dacd6fd8dc74eb1ee6852b2/library/core/src/ptr/unique.rs#L113C18-L115)). Then `fn ptr` can be marked `const` without changing its implementation.

Reply:
Sorry, not sure why GH put this comment on an empty line but it's meant to go with the `ptr` and `non_null` functions.

Anyway, this change plus the safety comments Oli requested and we can merge this.